src/main/io/serial_1wire.c

```diff
@@ -125,8 +140,13 @@ static void gpioSetOne(uint16_t escIndex, GPIO_Mode mode) {
 #endif
 
 #ifdef STM32F10X
+#ifdef USE_PREP_VARS
 #define ESC_INPUT(escIndex)    gpioSetOne(escIndex, Mode_IPU)
 #define ESC_OUTPUT(escIndex)   gpioSetOne(escIndex, Mode_Out_PP)
+#else
+#define ESC_INPUT(escIndex)    gpio_set_mode(escHardware[escIndex].gpio, (1U << escHardware[escIndex].pinpos), Mode_IPU);
```

Comment:
i'm not opposed to doing it this way, but i had trouble getting this to work with the slightly-off timing. ill try it again. i agree, simplicity is better if it works w/o gpioSetOne